pkg/volume/source.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package volume
+
+import (
+	"github.com/pkg/errors"
+)
+
+// Source represents the source of a volume to mount.
+// Only one of its members may be specified.
+type Source struct {
+	// Secret represents a secret that should populate this volume.
+	Secret []map[string]string `yaml:"secret,omitempty"`
```

Comment:
>[]map[string]string [](start = 8, length = 19)

When I see the array of map, I notice the secret yaml is defined as a list of key:value pair.

  - name: mysecrets
    secret:
      - mysecret1: {{.Secrets.sampleSecret | b64enc}}
      - mysecret2: {{"this is a non encoded string" | b64enc}}

 Why we choose it rather than the dictionary like 

  - name: mysecrets
    secret:
      mysecret1: {{.Secrets.sampleSecret | b64enc}}
      mysecret2: {{"this is a non encoded string" | b64enc}}

Reply:
Yes I originally tried just using a map[string]string but that wasn't working for some reason so I used an array of maps instead as a workaround. I'll try the single map way again and try to make it work.